File: $_algotsu-kaizen/problemo-set/partition-array-such-that-maximum-difference-is-k.c
int compare(const void *a, const void *b) {
    return (*(int*)a - *(int*)b);
}

int partitionArray(int* nums, int numsSize, int k) {
    qsort(nums, numsSize, sizeof(int), compare);
    int result = 0;

    // for (int i = 0; i < numsSize; ++i) {
    //     printf(" %d ", nums[i]);
    // }

    int l = 0, r;
    while (l < numsSize) {
        r = l;
        while (r < (numsSize - 1) && nums[r + 1] - nums[l] <= k) ++r;
        l = r + 1;
        result++;
    }

    return result;
}

File: $_algotsu-kaizen/problemo-set/partition-array-such-that-maximum-difference-is-k.rs
impl Solution {
    pub fn partition_array(mut nums: Vec<i32>, k: i32) -> i32 {
        nums.sort();
        let mut result: i32 = 0;
        let (mut l, mut r): (usize, usize) = (0usize, 0usize);

        while l < nums.len() {
            r = l;
            while (r < (nums.len() - 1usize)) && (nums[r + 1] - nums[l] <= k) {
                r += 1;
            }

            l = r + 1;
            result += 1;
        }

        result
    }
}
